service-registry/registry-zero-config/src/test/java/org/apache/servicecomb/serviceregistry/client/TestZeroConfigRegistryClientImpl.java

```diff
@@ -0,0 +1,388 @@
+    }
+
+    @Test
+    public void test_isSchemaExist_whenServiceIdIsItselfId_shouldReturnTrue(){
+        ClientUtil.microserviceSelf = prepareService(selfServiceId, true);
+
+        boolean returnedResult = target.isSchemaExist(selfServiceId, schemaId1);
+        Assert.assertTrue(returnedResult);
+    }
+
+    @Test
+    public void test_isSchemaExist_whenSchemaNotExist_shouldReturnFalse(){
+        ClientUtil.microserviceSelf =  prepareService(selfServiceId, true);
+
+        boolean returnedResult = target.isSchemaExist(selfServiceId, nonExistSchemaId);
+        Assert.assertFalse(returnedResult);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void test_registerSchema_whenServiceIdIsNull_shouldThrowIllegalArgumentException(){
+        ClientUtil.microserviceSelf =  prepareService(selfServiceId, false);
+
+        target.registerSchema(selfServiceId, schemaId1, schemaContent1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void test_registerSchema_whenServiceIdIsNotItSelfId_shouldThrowIllegalArgumentException(){
+        ClientUtil.microserviceSelf =  prepareService(selfServiceId, true);
+
+        target.registerSchema(otherServiceId, schemaId1, schemaContent1);
+    }
+
+    @Test
+    public void test_registerSchema_whenReigsterForItself_shouldSuceed(){
+        ClientUtil.microserviceSelf =  prepareService(selfServiceId, true);
+
+        boolean returnedResult = target.registerSchema(selfServiceId, newSchemaId, newSchemaContent);
+        Assert.assertTrue(returnedResult);
+        Assert.assertEquals(newSchemaContent, ClientUtil.microserviceSelf.getSchemaMap().computeIfPresent(newSchemaId, (k, v) -> {
+            return v;
+        }));
+    }
+
+    @Test
+    public void test_getSchema_whenForSelfMicroservice_shouldSuceed(){
+        ClientUtil.microserviceSelf = prepareService(selfServiceId, true);
+
+        String returnedSchemaContent = target.getSchema(selfServiceId, schemaId1);
+        Assert.assertEquals(schemaContent1, returnedSchemaContent);
+    }
+
+    @Test
+    public void test_getSchema_whenForSelfMicroservice_shouldNotCallZeroConfigRegistryServiceAndSucceed(){
+        ClientUtil.microserviceSelf = prepareService(selfServiceId, true);
+
+        String returnedSchemaContent = target.getSchema(selfServiceId, schemaId1);
+
+        Assert.assertEquals(schemaContent1, returnedSchemaContent);
+        verifyZeroInteractions(zeroConfigRegistryService);
+    }
+
+    @Test
+    public void test_getSchema_whenForOtherMicroservice_shouldCallZeroConfigRegistryService(){
+        ClientUtil.microserviceSelf = prepareService(selfServiceId, true);
+        when(zeroConfigRegistryService.getMicroservice(otherServiceId)).thenReturn(prepareServerServiceInstance(true));
+        String schemaContentEndpoint = endpoint1 + SCHEMA_CONTENT_ENDPOINT_BASE_PATH + SCHEMA_CONTENT_ENDPOINT_SUBPATH + "?" + SCHEMA_CONTENT_ENDPOINT_QUERY_KEYWORD + "=" + schemaId1;
```

Comment:
This line is too long. use etc/*xml code template to format files.

Reply:
after removing those "long" constant， now. it's much shorter:)